### ISSUE TITLE
iOS initPaymentSheet crash

### ISSUE THREAD

**Issue by nardi-xhepi** [flutter-stripe/flutter_stripe#1407]
**Describe the bug**
When the app calls "Stripe.instance.initPaymentSheet" function, the iOS version of the app crashes while everything works good on android. 

The error I get is: 

```
12:32:44 I Runner(libswiftCore.dylib)[526] <Notice>: Could not cast value of type 'NSNull' (0x203232f78) to 'NSDictionary' (0x203234888).
``` 

**To Reproduce**
Here is what I use to initialize the payment sheet: 

```dart
await Stripe.instance.initPaymentSheet(
        paymentSheetParameters: SetupPaymentSheetParameters(
          merchantDisplayName: 'name',
          customerEphemeralKeySecret: data?['ephemeralKey'],
          customerId: data?['customer'],
          paymentIntentClientSecret: paymentIntent,
          googlePay: PaymentSheetGooglePay(
            merchantCountryCode: 'fr',
          ),
          applePay: PaymentSheetApplePay(
            merchantCountryCode: 'fr',
          ),
          style: ThemeMode.dark,
        ),
      );
``` 

The "data" variable and the fields I'm retreiving from it are non null values.

**Smartphone / tablet**
 - Device: [iPhone XR]
 - OS: [iOS 16.3]
 - Package version: [9.4.0]
 - Flutter version [2.19.6]

**Comment by remonh87** [comment]
I think you have an older version of the stripe iOS library can you make sure stripe_ios is at version 9.4.1 ? Also if this doesn't resolve the issue can you provide me with a runnable example of an app that contains the problem? I cannot reproduce it with the code supplied in our example app

**Comment by nardi-xhepi** [comment]
I tried debugging it in XCode. I still get the same error: 
```
Could not cast value of type 'NSNull' (0x1ba22c150) to 'NSDictionary' (0x1ba22a900).
2023-09-13 11:06:18.337337+0200 Runner[55678:571321] Could not cast value of type 'NSNull' (0x1ba22c150) to 'NSDictionary' (0x1ba22a900).
``` 

And this is the line in StripePlugin that raises the error:

```swift
let mutable = (params["intentConfiguration"] as! NSDictionary).mutableCopy() as! NSMutableDictionary
```

**Comment by nardi-xhepi** [comment]
After several tries and debugging, I finally found a solution. In my app, I don't have to use "intentConfiguration" parameter in Stripe.initPayementSheet. Therefore, I deleted this code snippet from StripePlugin.swift in stripe_ios: 

```swift
if (params.object(forKey: "intentConfiguration") != nil) {
            let mutable = (params["intentConfiguration"] as! NSDictionary).mutableCopy() as! NSMutableDictionary
            mutable["confirmHandler"] = true;
            let adjusted = params.mutableCopy() as! NSMutableDictionary
            adjusted["intentConfiguration"] = mutable
            params = adjusted
        }
``` 

I hope this helps anyone encountering the same error.

**Comment by jonasbark** [comment]
@nardi-xhepi you're looking at old code. It has already been fixed with stripe_ios 9.4.1
https://github.com/flutter-stripe/flutter_stripe/blob/main/packages/stripe_ios/ios/Classes/StripePlugin.swift#L183

**Comment by nardi-xhepi** [comment]
Awsome !
